fix(popups): do not override popup size from window features

dgozman opened the pull request:
We usually force window size from the browser context. However, popups that have window features insist on a specific window size, so we respect that.

pavelfeldman commented on src/chromium/crPage.ts:
```diff
@@ -68,6 +69,12 @@ export class CRPage implements PageDelegate {
+      const features = opener._popupFeatures.shift() || [];
+      const viewportSize = helper.getViewportSizeFromWindowFeatures(features);
+      if (viewportSize)
+        this._page._state.viewportSize = viewportSize;
```
why not viewportSize === null?

pavelfeldman commented on src/chromium/crPage.ts:
```diff
@@ -429,7 +437,7 @@ class FrameSession {
       promises.push(this._client.send('Page.setBypassCSP', { enabled: true }));
     if (options.ignoreHTTPSErrors)
       promises.push(this._client.send('Security.setIgnoreCertificateErrors', { ignore: true }));
-    if (this._isMainFrame() && options.viewport)
```
Let's make sure that we don't run updateViewport in case there is no viewport?

dgozman replied:
Done.

pavelfeldman commented on src/chromium/crPage.ts:
```diff
@@ -706,23 +718,21 @@ class FrameSession {
+    const viewportSize = this._page._state.viewportSize || { width: 0, height: 0 };
+    const isLandscape = viewportSize.width > viewportSize.height;
     const promises = [
       this._client.send('Emulation.setDeviceMetricsOverride', {
```
Let's make sure we don't touch it when there is no viewport.

pavelfeldman commented on src/webkit/wkPage.ts:
```diff
@@ -573,24 +573,21 @@ export class WKPage implements PageDelegate {
 
   async _updateViewport(): Promise<void> {
     const options = this._browserContext._options;
-    let viewport = options.viewport || { width: 0, height: 0 };
```
We want this code to stay for Joel's setViewport on the browser context.

dgozman committed:
fix(popups): do not override popup size from window features

We usually force window size from the browser context. However,
popups that have window features insist on a specific window size,
so we respect that.
